src/item.cpp

```diff
@@ -578,8 +578,14 @@ bool item::stacks_with( const item &rhs ) const
     }
     if( goes_bad() ) {
         // If this goes bad, the other item should go bad, too. It only depends on the item type.
-        if( bday != rhs.bday ) {
-            return false;
+        // Stack items that fall into the same "bucket" of freshness.
+        // Distant buckets are larger than near ones.
+        std::pair<int, clipped_unit> my_clipped_time_to_rot =
+            clipped_time( type->comestible->spoils - rot );
+        std::pair<int, clipped_unit> other_clipped_time_to_rot =
+            clipped_time( rhs.type->comestible->spoils - rhs.rot );
+        if( my_clipped_time_to_rot == other_clipped_time_to_rot ) {
+            return true;
```

Comment:
Items belonging to the same bucket may belong to different buckets after their rot timer advances, which might cause erratic stacking of items. For example, if two items have perish time values of 1.8 hours and 1.2 hours, after 0.3 hours the values becomes 1.5 hours and 54 minutes, and then the two items should not stack.

Also, better test `my_clipped_time_to_rot != other_clipped_time_to_rot` and return false if it's true, since there are other tests below. And the test `std::abs( to_turns<int>( rot - rhs.rot ) ) > processing_speed()` below probably should be removed.

Reply:
I need to add some tests for this, but I believe they will split and rejoin when necessary, and I observed this happening in practice when manually testing.

Point taken with !=, I had a different solution in place and ended up here via refactoring.

You're right about the processing speed thing, it's no longer necessary either.